Lib/ast.py

```diff
@@ -551,6 +553,671 @@ def __new__(cls, *args, **kwargs):
     type(...): 'Ellipsis',
 }
 
+# Large float and imaginary literals get turned into infinities in the AST.
+# We unparse those infinities to INFSTR.
+INFSTR = "1e" + repr(sys.float_info.max_10_exp + 1)
+
+def interleave(inter, f, seq):
+    """Call f on each item in seq, calling inter() in between.
+    """
+    seq = iter(seq)
+    try:
+        f(next(seq))
+    except StopIteration:
+        pass
+    else:
+        for x in seq:
+            inter()
+            f(x)
+
+class _Unparser:
```

Comment:
Why this is a private class? People should subclass it and alter things, add new nodes etc. Like tokenize.Untokenizer IMHO this should be public

Reply:
I don't feel comfortable to expose the class to be honest. The interface that I proposed to expose is `unparse()`, not the whole class. Notice that the class is not even an AST visitor and many of the node managing functions are private.